File: C++/Conditional/SwitchCase.cpp
#include <iostream>
using namespace std;

int main() {
    // Switch-Case Example

    int day = 3;
    string dayName;

    // Example 1: Simple switch-case statement
    switch (day) {
        case 1:
            dayName = "Monday";
            break;
        case 2:
            dayName = "Tuesday";
            break;
        case 3:
            dayName = "Wednesday";
            break;
        case 4:
            dayName = "Thursday";
            break;
        case 5:
            dayName = "Friday";
            break;
        default:
            dayName = "Weekend day";
    }

    // Example 2: Using switch-case for character comparison
    char grade = 'B';
    string gradeDescription;

    switch (grade) {
        case 'A':
        case 'a':  // Handling both uppercase and lowercase
            gradeDescription = "Excellent";
            break;
        case 'B':
        case 'b':
            gradeDescription = "Good";
            break;
        case 'C':
        case 'c':
            gradeDescription = "Average";
            break;
        case 'D':
        case 'd':
            gradeDescription = "Poor";
            break;
        case 'F':
        case 'f':
            gradeDescription = "Fail";
            break;
        default:
            gradeDescription = "Invalid grade";
    }

    // Output results
    cout << "Example 1: Day " << day << " is " << dayName << endl;
    cout << "Example 2: Grade " << grade << " is " << gradeDescription << endl;

    return 0;
}

File: Lua/Conditional/SwitchCase.lua
-- Switch Case

local dayOfWeek = 3  -- Representing Wednesday

local daySwitch = {
    [1] = function() print("Today is Monday") end,
    [2] = function() print("Today is Tuesday") end,
    [3] = function() print("Today is Wednesday") end,
    [4] = function() print("Today is Thursday") end,
    [5] = function() print("Today is Friday") end,
    [6] = function() print("Today is Saturday") end,
    [7] = function() print("Today is Sunday") end
}

local caseFunction = daySwitch[dayOfWeek]
if caseFunction then
    caseFunction()
else
    print("Invalid day of the week")
end
